lib/promscrape/discovery/puppetdb/resource.go

```diff
@@ -0,0 +1,170 @@
+package puppetdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promutils"
+)
+
+const (
+	separator = ","
+)
+
+var matchContentTypeRegex = regexp.MustCompile(`^(?i:application\/json(;\s*charset=("utf-8"|utf-8))?)$`)
+
+type resource struct {
+	Certname    string     `json:"certname"`
+	Resource    string     `json:"resource"`
+	Type        string     `json:"type"`
+	Title       string     `json:"title"`
+	Exported    bool       `json:"exported"`
+	Tags        []string   `json:"tags"`
+	File        string     `json:"file"`
+	Environment string     `json:"environment"`
+	Parameters  parameters `json:"parameters"`
+}
+
+type parameters map[string]interface{}
+
+// toLabels convert Parameters map into label-value map.
+// See: https://github.com/prometheus/prometheus/blob/685493187ec5f5734777769f595cf8418d49900d/discovery/puppetdb/resources.go#L39
+func (p *parameters) toLabels() map[string]string {
```

Comment:
WDYT of additing new labels directly instead of allocating additional map? 

```go
 func (p *parameters) addToLabels(keyPrefix string, m *promutils.Labels) {
}
....

	if cfg.includeParameters {
                   resource.Parameters.addToLabels("__meta_puppetdb_parameter_", m)
		}

```